docs/proposals/13-automated-seed-management.md

```diff
@@ -0,0 +1,280 @@
+
+To create, delete, and update seeds effectively in a declarative way and allow auto-scaling, a "creatable seed" resource along with a "set" (and in the future, perhaps also a "deployment") of such creatable seeds should be introduced, similar to Kubernetes `Pod`, `ReplicaSet`, and `Deployment` (or to MCM `Machine`, `MachineSet`, and `MachineDeployment`) resources. With such resources (and their respective controllers), creating a new seed based on a template would become as simple as increasing the `replicas` field in the "set" resource. 
+
+In [Issue #2181](https://github.com/gardener/gardener/issues/2181) it is already proposed that the `use-as-seed` annotation is replaced by a dedicated `ShootedSeed` resource. The solution proposed here further elaborates on this idea.
+
+### ShootedSeeds
+
+#### ShootedSeed Resource
+
+The `ShootedSeed` resource is a dedicated custom resource that represents a "shooted seed" and properly replaces the `use-as-seed` annotation. This resource contains:
+
+* A Shoot template that contains the Shoot spec and parts of the metadata, such as labels and annotations.
+* A Seed template that contains the Seed spec and parts of the metadata, such as labels and annotations.
+* A `gardenlet` section that contains:
+    * Whether `gardenlet` is enabled or not.
+    * Certain aspects of the `gardenlet` deployment configuration, such as the number of replicas, the image, which bootstrap mechanism to use (bootstrap token / service account), etc.
+    * The `GardenletConfiguration` resource that contains controllers configuration, feature gates, etc.
+    
+A ShootedSeed allows fine-tuning the seed and the `gardenlet` configuration of shooted seeds in order to deviate from the global defaults, e.g. lower the concurrent sync for some of the seed's controllers or enable a feature gate only on certain seeds. Also, it could simplify the deletion protection of shooted seeds. Last but not least, ShootedSeeds could be used as the basis for creating and deleting seeds automatically via the `ShootedSeedSet` resource that is described in more details below.
+
+Unlike the `Seed` resource, the `ShootedSeed` resource is namespaced. If created in the `garden` namespace, the resulting seeds are globally available. If created in a project namespace, the resulting seeds can be used as "private seeds" by shoots in the project, either by being decorated with project-specific taints and labels, or by being of the special `PrivateSeed` kind that is also namespaced. The concept of private seeds / cloudprofiles is described in [Issue #2874](https://github.com/gardener/gardener/issues/2874). Until this concept is implemented, `ShootedSeed` resources might need to be restricted to the `garden` namespace, similarly to how shoots with the `use-as-seed` annotation currently are.
```

Comment:
It seems that there is a potential similarity of nomenclature between `Seed` and `Role` as scope is introduced - which is to ask if it is better to use `ClusterSeed` and `Seed` instead of `Seed` and `PrivateSeed`. The downside is this would be a very intensive upgrade.